homeassistant/components/plex/__init__.py

```diff
@@ -156,8 +159,8 @@ def _setup_plex(hass, config):
     """Unload a config entry."""
     server_id = entry.data[CONF_SERVER_IDENTIFIER]
 
-    cancel = hass.data[PLEX_DOMAIN][REFRESH_LISTENERS].pop(server_id)
-    cancel()
+    ws = hass.data[PLEX_DOMAIN][WEBSOCKETS].pop(server_id)
+    ws.close()
 
     dispatchers = hass.data[PLEX_DOMAIN][DISPATCHERS].pop(server_id)
     for unsub in dispatchers:
```

Comment:
The dispatchers unsub appended from the server are not async safe. We can't mix sync with async function to unsubscribe. At least not if they should all be unsubscribed from the same place and context. If they are all sync we can schedule that work on the executor thread pool.